OADP-2130: Restructure of the OADP Introduction (#64198)
openshift/openshift-docs: backup_and_restore/application_backup_and_restore/oadp-intro.adoc

:_content-type: ASSEMBLY
[id="oadp-introduction"]

> include::_attributes/common-attributes.adoc[]

Reviewer: We should reorganize this metadata info so that it follows OCP guidelines: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#assembly-file-metadata

```suggestion
:_content-type: ASSEMBLY
[id="oadp-introduction"]
= Introduction to {oadp-full}
include::_attributes/common-attributes.adoc[]
:context: oadp-api
:namespace: openshift-adp
:local-product: OADP

toc::[]
```

= Introduction to {oadp-full}
:context: oadp-api
:namespace: openshift-adp
:local-product: OADP

toc::[]

> The {oadp-first} product safeguards customer applications on {product-title}. OADP provides disaster recovery protection for {product-title} applications, application-related cluster resources, persistent volumes, and internal images.

Reviewer: Gonna take a stab and rewriting/combining these first two/three paragraphs: 

```suggestion
The {oadp-first} product safeguards customer applications on {product-title}. It offers comprehensive disaster recovery protection, covering OpenShift Container Platform applications, application-related cluster resources, persistent volumes, and internal images. OADP is also capable of backing up both containerized applications and virtual machines (VMs). However, it does not serve as a disaster recovery solution for etcd or OpenShift Operators. 
```

Author: Thanks for the re-work

I would like to keep the mention of etc highlighted, using whitespace. So I am going to drop that line down.

OADP can be used to back up containerized applications and virtual machines (VMs).

> OADP cannot be used as a disaster recovery solution for etcd or OpenShift Operators. For more information on etcd, see xref:../../backup_and_restore/control_plane_backup_and_restore/backing-up-etcd.adoc#backup-etcd[Backing up etcd].

Reviewer: ". . . information on" is used, but ". . . information about" is more commonly used in our docs:

I have a few suggestions here for how better to word this for clarity, because I don't think the link necessarily provides information about "etcd", but rather "backing up etcd".

1:
```suggestion
OADP cannot be used as a disaster recovery solution for etcd or OpenShift Operators. See the documentation for xref:../../backup_and_restore/control_plane_backup_and_restore/backing-up-etcd.adoc#backup-etcd[Backing up etcd] for more information. 
```
2:
```suggestion
For more information about backing up etcd, see xref:../../backup_and_restore/control_plane_backup_and_restore/backing-up-etcd.adoc#backup-etcd[Backing up etcd].
```

That said, I'm not really crazy about including the link to this information right here. I think that it could go on the bottom of the page, after the content, in an additional resources section. See comment below for example. 

Kind of up to you how you want to proceed.

> 

Reviewer: I'm kind of playing around with the idea of combining those three paragraphs (as I did above), and then making a separate section for the APIs.  eg:

```suggestion
// Module included in the following assemblies:
//
// * backup_and_restore/application_backup_and_restore/oadp-intro.adoc

[id="oadp-apis_{context}"]
= {oadp-full} APIs

{oadp-first} provides APIs to enable multiple approaches to customize backups and prevent the inclusion of unnecessary or inappropriate resources. 

OADP provides the following APIs:

* xref:../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/backing-up-applications.adoc#backing-up-applications[Backup]
* xref:../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/restoring-applications.adoc#restoring-applications[Restore]
* Schedule
* BackupStorageLocation
* VolumeSnapshotLocation

```

OADP provides APIs to enable multiple approaches to customize a backup to avoid backing up inappropriate resources.

OADP provides the following APIs:

* xref:../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/backing-up-applications.adoc#backing-up-applications[Backup]
* xref:../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/restoring-applications.adoc#restoring-applications[Restore]
* Schedule
* BackupStorageLocation
* VolumeSnapshotLocation

> 

Reviewer: ```suggestion
[role="_additional-resources"]
.Additional resources

* xref:../../backup_and_restore/control_plane_backup_and_restore/backing-up-etcd.adoc#backup-etcd[Backing up etcd]
```

> * xref:../../backup_and_restore/application_backup_and_restore/oadp-api.adoc#oadp-api[Schedule]

Reviewer 2: this link should redirect to the Scheduling Backups section.
https://64198--docspreview.netlify.app/openshift-enterprise/latest/backup_and_restore/application_backup_and_restore/oadp-api#oadp-api

I dont see anything related to Schedule on the current link it redirects to.

Author: No

Author: @weshayutin && @akarol && @stillalearner - let me explain - I am working mostly on structural changes, once I have the structure changed I think it will be easier to add re-writes. 
Please can I just get a LGTM and I can get this moving

:_content-type: ASSEMBLY
[id="oadp-introduction"]
:_content-type: ASSEMBLY
[id="oadp-introduction"]
= Introduction to {oadp-full}
include::_attributes/common-attributes.adoc[]
:context: oadp-api
:namespace: openshift-adp
:local-product: OADP

toc::[]
= Introduction to {oadp-full}
:context: oadp-api
:namespace: openshift-adp
:local-product: OADP

> 

Reviewer 3: Why is the [title and metadata repeated](https://github.com/openshift/openshift-docs/pull/64198/files#diff-1ee50b000b47077286254c47ff98fdc2505e4a0b5f2b7c2d4b464d8816860ccfR5-R15)?

toc::[]

> The {oadp-first} product safeguards customer applications on {product-title}. It offers comprehensive disaster recovery protection, covering OpenShift Container Platform applications, application-related cluster resources, persistent volumes, and internal images. OADP is also capable of backing up both containerized applications and virtual machines (VMs). 

Reviewer 3: ```suggestion
The {oadp-first} product safeguards customer applications on {product-title}. It offers comprehensive disaster recovery protection, covering {product-title} applications, application-related cluster resources, persistent volumes, and internal images. OADP is also capable of backing up both containerized applications and virtual machines (VMs). 
```

However, OADP does not serve as a disaster recovery solution for xref:../../backup_and_restore/control_plane_backup_and_restore/backing-up-etcd.adoc#backup-etcd[etcd] or OpenShift Operators. 

[id="oadp-apis_{context}"]
= {oadp-full} APIs

> {oadp-first} provides APIs to enable multiple approaches to customize backups and prevent the inclusion of unnecessary or inappropriate resources. 

Reviewer 3: ```suggestion
{oadp-first} provides APIs that enable multiple approaches to customizing backups and preventing the inclusion of unnecessary or inappropriate resources. 
```

OADP provides the following APIs:

* xref:../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/backing-up-applications.adoc#backing-up-applications[Backup]
* xref:../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/restoring-applications.adoc#restoring-applications[Restore]
* Schedule
* BackupStorageLocation
* VolumeSnapshotLocation

OADP provides APIs to enable multiple approaches to customize a backup to avoid backing up inappropriate resources.

OADP provides the following APIs:

* xref:../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/backing-up-applications.adoc#backing-up-applications[Backup]
* xref:../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/restoring-applications.adoc#restoring-applications[Restore]
* Schedule
* BackupStorageLocation

> * VolumeSnapshotLocation

Reviewer 3: Why is [this text repeated](https://github.com/openshift/openshift-docs/pull/64198/files#diff-1ee50b000b47077286254c47ff98fdc2505e4a0b5f2b7c2d4b464d8816860ccfR40-R46)?

Author: Cos I made a mistake

> [role="_additional-resources"]

Reviewer 3: Need a carrieage return before the `[role="_additional-resources"]`.

Author: ta muchly
